ttnn/cpp/ttnn/operations/ccl/common/host/ccl_worker_builder.cpp

```diff
@@ -0,0 +1,1575 @@
+        args.push_back(sync_details.value().num_signals);
+        for (size_t i = 0; i < sync_details.value().num_signals; ++i) {
+            auto const noc_coord =
+                this->device->virtual_core_from_logical_core(sync_details.value().core, CoreType::WORKER);
+            log_trace(
+                tt::LogOp,
+                "ccl_send_writer on device {} adding sync signal dest to (y={},x={},id={})",
+                this->device->id(),
+                noc_coord.y,
+                noc_coord.x,
+                sync_details.value().sem_ids[i]);
+            args.push_back(sync_details.value().sem_ids[i]);
+            args.push_back(noc_coord.x);
+            args.push_back(noc_coord.y);
+        }
+    }
+
+    auto const& addr_gen_rt_args = ttnn::ccl::emit_address_generator_runtime_args(this->device, output_tensor);
+    std::ranges::copy(addr_gen_rt_args, std::back_inserter(args));
+    for (auto const& arg : addr_gen_rt_args) {
+        log_trace(tt::LogOp, "ccl_send_writer arg[{}]: addr_gen_rt_args[] {}", logged_arg_idx, args[logged_arg_idx]);
+        logged_arg_idx++;
+    }
+
+    log_trace(tt::LogOp, "ccl_send_writer Generating {} ccl send commands", slices.size());
+    emit_ccl_send_slice_sequence_commands(slices, args);
+
+    log_trace(tt::LogOp, "ccl_send_writer Sender Worker has {} RT Args: {}", args.size(), args);
+
+    return args;
+}
+
+std::vector<uint32_t> CCLWorkerArgBuilder::generate_sender_reader_kernel_ct_args() const {
+    auto const& input_tensor = this->op_config.get_input_tensor(0);
+    std::vector<uint32_t> args = {
+        static_cast<uint32_t>(input_tensor.memory_config().memory_layout),  // tensor memory layout
+        static_cast<uint32_t>(input_tensor.buffer()->buffer_type()),        // buffer type
+        static_cast<uint32_t>(input_tensor.layout()),                       // page layout
+        static_cast<uint32_t>(tt::CB::c_in0)                                // cb_id
+    };
+
+    auto const& addr_gen_rt_args = ttnn::ccl::emit_address_generator_compile_time_args(input_tensor);
+    std::ranges::copy(addr_gen_rt_args, std::back_inserter(args));
+
+    return args;
+}
+
+std::vector<uint32_t> CCLWorkerArgBuilder::generate_sender_writer_kernel_ct_args() const {
+    auto const& output_tensor = this->op_config.get_output_tensor(0);
+    std::vector<uint32_t> args = {
+        static_cast<uint32_t>(output_tensor.memory_config().memory_layout),  // tensor memory layout
+        static_cast<uint32_t>(output_tensor.buffer()->buffer_type()),        // buffer type
+        static_cast<uint32_t>(output_tensor.layout()),                       // page layout
+        static_cast<uint32_t>(tt::CB::c_in0)                                 // cb_id
+    };
+
+    auto const& addr_gen_rt_args = ttnn::ccl::emit_address_generator_compile_time_args(output_tensor);
+    std::ranges::copy(addr_gen_rt_args, std::back_inserter(args));
+
+    return args;
+}
+
+}  // namespace worker_detail
```

Comment:
Can these 3 be collapsed into ttnn::ccl::worker_detail since they start and end in the same place